### PR TITLE
Test shader uniforms need descriptor set and binding

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: libshaderc/src/shaderc_cpp_test.cc [hunk 1]
```diff
@@ -1414,6 +1420,9 @@ TEST_F(CppInterface, HlslFunctionality1SurvivesCloning) {
   CompileOptions options;
   options.SetSourceLanguage(shaderc_source_language_hlsl);
   options.SetHlslFunctionality1(true);
+  // The counter needs a binding, and there is no way to set it in the shader
```
Comment (dj2) [inline libshaderc/src/shaderc_cpp_test.cc]: Is there a glslang bug associated we could link so we can remove this when vk::counter_binding is supported?

File: libshaderc/src/shaderc_test.cc [hunk 2]
```diff
@@ -1821,7 +1827,7 @@ TEST_F(CompileStringWithOptionsTest, HlslFlexibleMemoryLayoutAllowed) {
                                               shaderc_source_language_hlsl);
   shaderc_compile_options_set_optimization_level(
       options_.get(), shaderc_optimization_level_performance);
-
+  shaderc_compile_options_set_auto_bind_uniforms(options_.get(), true);
```
Comment (dj2) [inline libshaderc/src/shaderc_test.cc]: no comment?

File: libshaderc/src/shaderc_cpp_test.cc (hunk 1 above)
Comment (dneto0) [reply libshaderc/src/shaderc_cpp_test.cc]: I just made one: https://github.com/KhronosGroup/glslang/issues/1616

Comment (dneto0) [reply libshaderc/src/shaderc_cpp_test.cc]: Added comment to point to the Glslang enhancement request

**Comment by dneto0** [comment]
Feedback addressed. PTAL

**Comment by dneto0** [comment]
Rebased.  Keeping the LGTM

**Comment by dneto0** [comment]
I can't reproduce the VS2017 failure on a debug build.

**Comment by dneto0** [comment]
The VS2013 issue has been fixed: https://github.com/KhronosGroup/glslang/issues/1630

**Pull request merged**